pontoon/base/views.py

```diff
@@ -988,50 +994,119 @@ def transvision(request):
         return HttpResponse("error")
 
 
+def download_file(prefixes, dirnames, relative_path):
+    for prefix in prefixes:
+        for dirname in dirnames:
+            url = os.path.join(prefix.replace('{locale_code}', dirname), relative_path)
+            r = requests.get(url, stream=True)
+            if not r.ok:
+                continue
+
+            filename = str(uuid.uuid4()) + os.path.splitext(url)[1]
+            with codecs.open(filename, 'wb') as destination:
+                for chunk in r.iter_content(chunk_size=1024):
+                    if chunk:
+                        destination.write(chunk)
+            return filename
+
+
+def get_download_content(slug, code, relative_path):
+    """
+    Get content of the file to be downloaded.
+
+    :param str slug: Project slug.
+    :param str code: Locale code.
+    :param str relative_path: Resource path.
+    """
+    project = get_object_or_404(Project, slug=slug)
+    locale = get_object_or_404(Locale, code__iexact=code)
+    resource = get_object_or_404(Resource, project__slug=slug, path=relative_path)
+
+    # Get locale file
+    locale_prefixes = (
+        project.repositories.filter(permalink_prefix__contains='{locale_code}')
+        .values_list('permalink_prefix', flat=True)
+        .distinct()
+    )
+    dirnames = set([locale.code, locale.code.replace('-', '_')])
+    locale_path = download_file(locale_prefixes, dirnames, relative_path)
+    if not locale_path:
+        return False
+
+    # Get source file if needed
+    source_path = None
+    if resource.is_asymmetric:
+        source_prefixes = (
+            project.repositories
+            .values_list('permalink_prefix', flat=True)
+            .distinct()
+        )
+        dirnames = VCSProject.SOURCE_DIR_NAMES
+        source_path = download_file(source_prefixes, dirnames, relative_path)
+        if not source_path:
+            return False
+
+    # Update file from database
+    resource_file = formats.parse(locale_path, source_path)
+    entities = {}
+
+    for e in Entity.objects.filter(
```

Comment:
I like to store complex querysets that I want to iterate over in a variable so that the for loop is easier to read.